tools/upload_text.php

```diff
@@ -249,164 +217,138 @@
     // make reasonably sure script does not timeout on large file uploads
     set_time_limit(14400);
 
-    // if files have been uploaded, process them and mangle the postcomments
-    $uploaded_file = validate_uploaded_file();
+    slim_header($title);
+    echo "<h1>$title</h1>";
+    echo "<h2>", sprintf("Project: %s", $project->nameofwork), "</h2>";
 
-    $have_file = FALSE;
-    if (is_file($uploaded_file))
+    // if files have been uploaded, process them and mangle the postcomments
+    $temporary_path = "";
+    $returning_to_pool = ('return_1' == $stage || 'return_2' == $stage);
+    try
     {
-        // replace filename
-        $zipext = ".zip";
-        $name = $projectid.$indicator.$zipext;
-        $location = "$project->dir/$name";
-        ensure_path_is_unused( $location );
-        copy($uploaded_file, $location);
-        unlink($uploaded_file);
-        $have_file = TRUE;
-        if ($stage == 'smooth_avail')
+        try
```

Comment:
Consider putting all of this inner-try logic into its own function. I think it's a fairly intact piece of code. That function thus either works or raises an exception. I think it only needs to be passed in `$returning_to_pool` and could return `$have_file`.